### PR TITLE
Remove Sphinx as a required dependency

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

**Comment by andy-maier** [comment]
Gentle reminder to get this in. We are also hampered by the circular dependency (using pipdeptree).

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Release 1.0.5 (Pending)
+==========================
```
Comment (AA-Turner) [inline CHANGES]:
```suggestion
Release 1.0.5 (unreleased)
==========================
```

File: pyproject.toml [hunk 2]
```diff
@@ -53,6 +50,9 @@ lint = [
     "mypy",
     "docutils-stubs",
 ]
+standalone = [
+  "sphinx > 5",
```
Comment (AA-Turner) [inline pyproject.toml]:
```suggestion
    "Sphinx>=5",
```

File: CHANGES [hunk 3]
```diff
@@ -1,3 +1,8 @@
+Release 1.0.5 (unreleased)
+==========================
+
+* Fix circular dependency with Sphinx that caused failure in DAG-based package management
```
Comment (AA-Turner) [inline CHANGES]:
```suggestion
* Remove Sphinx as a required dependency, as circular dependencies may cause
  failure with package managers that expect a directed acyclic graph (DAG)
  of dependencies.
```

File: pyproject.toml [hunk 4]
```diff
@@ -39,9 +39,6 @@ classifiers = [
     "Topic :: Text Processing",
     "Topic :: Utilities",
 ]
-dependencies = [
-    "Sphinx>=5",
-]
 dynamic = ["version"]
```
Comment (AA-Turner) [inline pyproject.toml:43]:
```suggestion
dependencies = []
dynamic = ["version"]
```

**Comment by jnikula** [comment]
The approach taken here (and perhaps since commit 448fbf6ef773 ("Support Python 3.9 and later")) makes it unnecessarily difficult to install or use Sphinx versions prior to 5. Sphinx since very early versions depends on sphinxcontrib-applehelp and sphinxcontrib-devhelp etc. without versions. Installing Sphinx < 5 pulls in latest sphinxcontrib-applehelp and sphinxcontrib-devhelp etc., because they satisfy the requirements. Then when you try to `sphinx-build`, you hit `The sphinxcontrib.devhelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.`

The only way to use Sphinx < 5 now is to figure out the versions of sphinxcontrib-applehelp and sphinxcontrib-devhelp and sphinxcontrib.htmlhelp (and maybe more, still in progress) that do not depend on Sphinx >= 5, and specify them explicitly like `sphinxcontrib-devhelp==1.0.2` etc.

**Pull request merged**